server/src/main/java/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -2358,4 +2383,109 @@ public boolean isUsageHidden(IPAddressVO ip) {
     public static ConfigKey<Boolean> getSystemvmpublicipreservationmodestrictness() {
         return SystemVmPublicIpReservationModeStrictness;
     }
+
+    @Override
+    public boolean checkIfPublicIpAddressIsNotInQuarantineAndCanBeAllocated(IpAddress ip, Account newOwner) {
```

Comment:
the name suggest a validation/check. I think 
```suggestion
    public boolean canPublicIpAddressBeAllocated(IpAddress ip, Account newOwner) {
```